app/uk/gov/hmrc/economiccrimelevyregistration/controllers/RegistrationSubmittedController.scala

```diff
@@ -29,22 +30,42 @@ import javax.inject.{Inject, Singleton}
 class RegistrationSubmittedController @Inject() (
   val controllerComponents: MessagesControllerComponents,
   authorise: AuthorisedActionWithoutEnrolmentCheck,
-  view: RegistrationSubmittedView
+  view: RegistrationSubmittedView,
+  bookmarkedView: BookmarkedRegistrationSubmittedView
 ) extends FrontendBaseController
     with I18nSupport {
 
   def onPageLoad: Action[AnyContent] = authorise { implicit request =>
-    val eclReference: String = request.session
-      .get(SessionKeys.EclReference)
-      .getOrElse(throw new IllegalStateException("ECL reference number not found in session"))
+    val sourceOfEclReference =
+      checkSourceOfEclReference(request.session.get(SessionKeys.EclReference), request.eclRegistrationReference)
 
-    val firstContactEmailAddress: String = request.session
-      .get(SessionKeys.FirstContactEmailAddress)
-      .getOrElse(throw new IllegalStateException("First contact email address not found in session"))
+    val eclReference: String = sourceOfEclReference match {
+      case "SESSION"   => request.session.get(SessionKeys.EclReference).get
+      case "ENROLMENT" => request.eclRegistrationReference.get
+      case _           => throw new IllegalStateException("ECL reference number not found in session or in enrolment")
+    }
+
+    val firstContactEmailAddress: String = sourceOfEclReference match {
+      case "SESSION"   => request.session.get(SessionKeys.FirstContactEmailAddress).get
+      case "ENROLMENT" => ""
+      case _           => throw new IllegalStateException("First contact email address not found in session")
+    }
 
     val secondContactEmailAddress: Option[String] = request.session.get(SessionKeys.SecondContactEmailAddress)
 
-    Ok(view(eclReference, firstContactEmailAddress, secondContactEmailAddress))
+    sourceOfEclReference match {
+      case "SESSION"   => Ok(view(eclReference, firstContactEmailAddress, secondContactEmailAddress))
+      case "ENROLMENT" => Ok(bookmarkedView(eclReference))
+    }
   }
 
+  private def checkSourceOfEclReference(sessionSource: Option[String], enrolmentSource: Option[String]): String =
+    sessionSource match {
+      case Some(_) => "SESSION"
```

Comment:
I would suggest making the "SESSION", "ENROLMENT" and "NONE" string constants private vals of the class then reference these in the code (reduces the risk of typos and makes it easier to change the values should we need to).

Reply:
Done 😄